### DISCUSSION TITLE
How to keep app only one instance?

### DISCUSSION THREAD

**Discussion by yaoyulian** [AvaloniaUI/Avalonia#9370] [Q&A] [answered]
if app is already running, click app icon will start run another app instance.  How to detect the same app is running in "MainWindow Initialize"?  Prompt “App is already running” with messagebox , then exit.

**Comment by jp2masa** [comment, marked as answer]
This is not Avalonia specific, you can use named mutexes or a lock file for example (this SO post may be helpful: https://stackoverflow.com/a/522874).
